lib/src/download_utils.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:http/http.dart' as http;
+import 'package:path/path.dart' as path;
+
+import 'logging.dart';
+import 'utils.dart';
+
+/// Returns a non-null Directory instance only if it is able to download and
+/// extract the direct package dependency. On any failure it clears the temp
+/// directory, otherwise it is the caller's responsibility to delete it.
+Future<Directory> downloadPackage(String package, String version) async {
+  final temp = await Directory.systemTemp.createTemp('pana-');
+  final dir = new Directory(await temp.resolveSymbolicLinks());
+  try {
+    final url =
+        'https://storage.googleapis.com/pub-packages/packages/$package-$version.tar.gz';
```

Comment:
Sure this handles weird URL characters that can be in versions?

You might consider using `new Uri.https`...

Reply:
ah, good point...